grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/GreedyTraversalPlan.java

```diff
@@ -221,10 +221,12 @@ private static void processFragmentWithFixedCost(List<Fragment> plan,
             plan.add(fragment);
             double logInstanceCount = -1D;
             if (fragment.getShardCount(tx).isPresent()) {
-                long shardCount = fragment.getShardCount(tx).get();
-                if (shardCount > 0) {
-                    logInstanceCount = Math.log(shardCount - 1D + SHARD_LOAD_FACTOR) +
-                            Math.log(tx.admin().shardingThreshold());
+                Optional<Long> shardCount = fragment.getShardCount(tx);
+                if(shardCount.isPresent()) {
```

Comment:
We're making this check twice now! I think you can remove the first check on line 223.